src/dex/maverick/maverick-integration.test.ts

```diff
@@ -0,0 +1,75 @@
+import dotenv from 'dotenv';
+dotenv.config();
+
+import { DummyDexHelper } from '../../dex-helper/index';
+import { Network, SwapSide } from '../../constants';
+import { BI_POWS } from '../../bigint-constants';
+import { Maverick } from './maverick';
+import {
+  checkPoolPrices,
+  checkPoolsLiquidity,
+  checkConstantPoolPrices,
+} from '../../../tests/utils';
+import { Tokens } from '../../../tests/constants-e2e';
+
+const network = Network.POLYGON;
+const TokenASymbol = 'WETH';
+const TokenA = Tokens[network][TokenASymbol];
+
+const TokenBSymbol = 'USDC';
+const TokenB = Tokens[network][TokenBSymbol];
+
+const amounts = [0n, 500000000000000000n, BI_POWS[18]];
+
+const dexKey = 'Maverick';
+
+describe('Maverick', function () {
+  it('getPoolIdentifiers and getPricesVolume SELL', async function () {
+    const dexHelper = new DummyDexHelper(network);
+    const blocknumber = await dexHelper.provider.getBlockNumber();
+    const maverick = new Maverick(network, dexKey, dexHelper);
+    await maverick.initializePricing(blocknumber);
+
+    const pools = await maverick.getPoolIdentifiers(
+      TokenA,
+      TokenB,
+      SwapSide.SELL,
+      blocknumber,
+    );
+    console.log(`${TokenASymbol} <> ${TokenBSymbol} Pool Identifiers: `, pools);
+
+    expect(pools.length).toBeGreaterThan(0);
+
+    const poolPrices = await maverick.getPricesVolume(
+      TokenA,
+      TokenB,
+      amounts,
+      SwapSide.SELL,
+      blocknumber,
+      pools,
+    );
+    console.log(`${TokenASymbol} <> ${TokenBSymbol} Pool Prices: `, poolPrices);
+
+    expect(poolPrices).not.toBeNull();
+    if (maverick.hasConstantPriceLargeAmounts) {
+      checkConstantPoolPrices(poolPrices!, amounts, dexKey);
+    } else {
+      checkPoolPrices(poolPrices!, amounts, SwapSide.SELL, dexKey);
+    }
```

Comment:
Please add `checkOnChainPricing` from [here](https://github.com/paraswap/paraswap-dex-lib/blob/5ee32b9c33545d0fa2c8bde3726a9ccb2f191f76/dex-template/__DexName__-integration.test.ts.template#L146). It seems, we added it after you started the integration process:

But it is a good indicator if there is an error in math or not :)